autodevops-create-db-secret/action.yml

```diff
@@ -0,0 +1,78 @@
+name: "Create database k8s secret"
+description: "Create random and secure database k8s secret if not already exists"
+inputs:
+  kubeconfig:
+    description: "The Kubernetes authentication configuration"
+  rancherId:
+    description: "The Rancher project ID, usually secrets.RANCHER_PROJECT_ID"
+  socialgouvBaseDomain:
+    description: "The base domain name, usually secrets.SOCIALGOUV_BASE_DOMAIN"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Get project and namespace names
+      shell: bash
+      run: |
+        if test -f ".github/dev.env"; then
+          cat ".github/dev.env" >> $GITHUB_ENV
+        fi
+        echo "namespace=${GITHUB_REPOSITORY#*/}-${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+    - name: Install kosko-charts autodevops
+      shell: bash
+      run: |
+        npx degit SocialGouv/kosko-charts/templates/autodevops /tmp/autodevops
+        yarn --cwd /tmp/autodevops
+
+    - name: Copy application k8s config to autodevops
+      shell: bash
+      run: |
+        if [ -f ".socialgouv/environments" ]; then
+          cp -r .socialgouv/environments /tmp/autodevops/
+        fi
+        cp .socialgouv/config.json /tmp/autodevops/
+
+    - name: Generate k8s namespace
+      shell: bash
+      run: |
+        yarn --cwd /tmp/autodevops --silent generate --env dev _namespace > namespace-${ENVIRONMENT}.yml
+      env:
+        ENVIRONMENT: ${{ inputs.environment }}
+        SOCIALGOUV_CONFIG_PATH: /tmp/autodevops/config.json
+        RANCHER_PROJECT_ID: ${{ inputs.rancherId || env.RANCHER_PROJECT_ID }}
+        SOCIALGOUV_BASE_DOMAIN: ${{ inputs.socialgouvBaseDomain || env.SOCIALGOUV_BASE_DOMAIN }}
+
+    - name: Get namespace name
+      uses: mikefarah/yq@v4.13.4
+      id: namespace
+      with:
+        cmd: yq eval '.metadata.name' namespace-${{ inputs.environment }}.yml
+
+    - name: Generate create-db-secret manifests
+      shell: bash
+      run: yarn --cwd /tmp/autodevops --silent generate --env dev jobs/create-db-secret > create-db-secret.yml
+      env:
+        SOCIALGOUV_CONFIG_PATH: /tmp/autodevops/config.json
+        RANCHER_PROJECT_ID: ${{ inputs.rancherId || env.RANCHER_PROJECT_ID }}
+        SOCIALGOUV_BASE_DOMAIN: ${{ inputs.socialgouvBaseDomain || env.SOCIALGOUV_BASE_DOMAIN }}
+
+    - name: Archive create-db-secret manifests
+      uses: actions/upload-artifact@v2
+      with:
+        name: create-db-secret.yml
+        path: create-db-secret.yml
+
+    - name: Create kubernetes config
+      shell: bash
+      run: |
+        mkdir ~/.kube
+        touch ~/.kube/config
+        echo ${{ inputs.kubeconfig }} | base64 -d > ~/.kube/config
+
+    - name: Create database secret if doesn't exists
+      shell: bash
+      run: kubectl apply --namespace=${{ steps.namespace.outputs.result }} -f create-db-secret.yml
```

Comment:
should we create the namespace if not exist before apply ?

Reply:
OK, good idea, je regarde comment mettre ça en place

Reply:
done with:
https://github.com/SocialGouv/actions/blob/06175a5d6837663113be992f5ab00a4c64791589/autodevops-create-db-secret/action.yml#L67-L70